### DISCUSSION TITLE
the function "forward" is called in function "optimize_parameters",is it all right?

### DISCUSSION THREAD

**Discussion by armst2016** [junyanz/pytorch-CycleGAN-and-pix2pix#1529] [Q&A] [answered]
as i known,"self.forward() "is called in "__call__()",there are some hooks in it。 we should avoid   calling Module.forward. But i find the function "forward" is called in function "optimize_parameters".  Are there some influences，result in it ?

**Comment by junyanz** [comment, marked as answer]
We are using BaseModel class [here](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/f13aab8148bd5f15b9eb47b690496df8dadbab0c/models/base_model.py#L8). It's not a Module class.
